packages/x-components/src/x-modules/search/store/actions/fetch-and-save-search-response.action.ts

```diff
@@ -33,17 +33,25 @@ const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
 function enrichRequest(request: InternalSearchRequest, state: SearchState): SearchRequest {
   const { page, ...restRequest } = request;
   const {
-    config: { pageSize },
+    config: { pageSize, pageMode },
     origin,
     results
   } = state;
-  const start = page === 1 ? 0 : results.length;
+
+  const start =
+    pageMode === 'infinite_scroll'
+      ? page === 1
+        ? 0
+        : results.length
+      : state.config.pageSize * page;
```

Comment:
Maybe it's me but IMO it's better to have an if else here rather than nested ternaries